zerver/tests/test_markdown.py

```diff
@@ -208,6 +208,21 @@ def markdown_convert_wrapper(content: str) -> str:
 
 
 class MarkdownMiscTest(ZulipTestCase):
+    def test_silent_mention(self) -> None:
+        # This test simply illustrates that we include Cordelia
+        # in mention_data.get_user_ids() even if the mention is
+        # silent. This may or may not have downstream implications
+        # in terms of us unnecessarily pushing out events related
+        # to the silent mention.  I don't believe this impacts
+        # the actual user. If we clean this up, this test will
+        # break in a hopefully good way, and we can clean up
+        # the assertion.
+        cordelia = self.example_user("cordelia")
+        mention_backend = MentionBackend(cordelia.realm_id)
+        content = f"@_**{cordelia.full_name}**"
+        mention_data = MentionData(mention_backend, content, message_sender=None)
+        self.assertEqual(mention_data.get_user_ids(), {cordelia.id})
```

Comment:
@abdelrahman725 does this test and comment seem useful to you? I think the current code is pretty clear about how silent mentions work.